apps/dcos_net/src/dcos_net_sup.erl

```diff
@@ -8,6 +8,11 @@ start_link() ->
     supervisor:start_link({local, ?MODULE}, ?MODULE, []).
 
 init([]) ->
+    try
```

Comment:
Here as well I need the `try` or some tests fail. I would rather have this initialization somewhere else where I would not need the try if possible. One other option is to move the try into the `init_metrics` function. What do you think?

Reply:
Please let's work on the code and the logic. Once it works, we can start discussing the tests.

Reply:
Ok, it's a tricky one. Here is a crash report:

```
=CRASH REPORT==== 4-Apr-2019::21:09:24.361147 ===
  crasher:
    initial call: application_master:init/4
    pid: <0.4144.0>
    registered_name: []
    exception exit: {{undef,
                         [{dcos_net_mesos_listener,init_metrics,[],[]},
                          {dcos_net_sup,init,1,
                              [{file,
                                   "/home/circleci/dcos-net/apps/dcos_net/src/dcos_net_sup.erl"},
                               {line,12}]},
                          {supervisor,init,1,
                              [{file,"supervisor.erl"},{line,295}]},
                          {gen_server,init_it,2,
                              [{file,"gen_server.erl"},{line,374}]},
                          {gen_server,init_it,6,
                              [{file,"gen_server.erl"},{line,342}]},
                          {proc_lib,init_p_do_apply,3,
                              [{file,"proc_lib.erl"},{line,249}]}]},
                     {dcos_net_app,start,[normal,[]]}}
      in function  application_master:init/4 (application_master.erl, line 138)
    ancestors: [<0.4143.0>]
    message_queue_len: 1
    messages: [{'EXIT',<0.4145.0>,normal}]
```

it means there is no `dcos_net_mesos_listener:init_metrics/0`.

Let's add `false = dcos_net_mesos_listener:module_info()` to this test:

```
{badmatch,
     [{module,dcos_net_mesos_listener},
      {exports,[{module_info,0},{module_info,1},{poll,0}]},
      {attributes,
          [{vsn,[289324849627647645634493986770560104722]},
           {lager_records,
               [{state,
                    [pid,ref,size,buf,timeout,timeout_ref,agents,frameworks,
                     tasks,waiting_tasks,subs]}]}]},
      {compile,[{version,"7.3.2"},{options,[]}]},
      {native,false},
      {md5,<<217,169,240,108,123,178,51,84,132,179,28,36,148,194,57,18>>}]}
```

So only `dcos_net_mesos_listener:poll/0` function is exported. That's definitely not true if you check the code of the module. It means it's a fake module. There is a test that is mocking the module, but doesn't call the corresponding `meck:unmock/1` function.